src/pages/download/_downloadPythonPackage.js

```diff
@@ -53,24 +54,103 @@ export async function generateZIP(deviceId) {
     // Only zip successfully generated mockups
     if (value !== null) {
       var file = dataURLtoFile(value, key.substring(3, key.length) + ".png");
-      images.set(key, URL.createObjectURL(file));
+      // images.set(key, URL.createObjectURL(file));
     } else {
       failedImages.push(key);
     }
   });
-  images.forEach(async function (imgURL, k) {
-    var filename = unescape(k.substring(3, k.length)) + ".png";
-    var image = await fetch(imgURL);
-    var imageBlob = await image.blob();
-    zip.file(filename, imageBlob, {
-      binary: true,
-      date: getJSZipDateWithOffset(),
-    });
-    count++;
-    if (count == images.size) {
-      zip.generateAsync({ type: "blob" }).then(function (content) {
-        saveAs(content, zipFilename);
+
+  // Download generated mockup if any mockup generated
+  if (images.size > 0) {
+    images.forEach(async function (imgURL, k) {
+      var filename = unescape(k.substring(3, k.length)) + ".png";
+      var image = await fetch(imgURL);
+      var imageBlob = await image.blob();
+      zip.file(filename, imageBlob, {
+        binary: true,
+        date: getJSZipDateWithOffset(),
       });
-    }
-  });
+      count++;
+      if (count == images.size) {
+        zip.generateAsync({ type: "blob" }).then(function (content) {
+          saveAs(content, zipFilename);
+        });
+      }
+    });
+  }
+
+  // Handle no generated mockup
+  if (failedImages.length > 0 && images.size > 0) {
+    const errorNode = document.createElement("div");
+    const errorHeader = document.createElement("div");
+    const errorMessage = document.createElement("div");
+
+    errorHeader.style.fontWeight = "bold";
+    errorHeader.style.color = "white";
+    errorHeader.innerHTML = "Partial Success";
+
+    errorMessage.innerHTML =
+      "<div>Below image(s) generated failed, Please try different image/device</div>";
+    errorMessage.innerHTML += "<ul>";
+    failedImages.forEach((failedImage) => {
+      errorMessage.innerHTML += `<li>${failedImage}</li>`;
+    });
+    errorMessage.innerHTML += "</ul><br>";
+    errorMessage.innerHTML +=
+      "<div>If it persists, we'd appreciate if you report it on our GitHub 🙏  <br><a href='https://github.com/oursky/mockuphone.com/issues'>https://github.com/oursky/mockuphone.com/issues.</a></div>";
```

Comment:
1. once we extract this as a `showToast` API, this should be much more readable.
2. Instead of `+=`, can we do one big `=` with string templating like `<ul>${myList}</ul>`?